lib/utapi/utilities.js

```diff
@@ -156,6 +156,20 @@ function listMetrics(metricType) {
         }
     });
 
+    // The string `commander[metric]` is a comma-separated list of resources
+    // given by the user.
+    const resources = commander[metric].split(',');
+
+    // Validate passed accounts to remove any canonicalids
+    if (metric === 'accounts') {
+        resources.forEach(r => {
+            if (!/^\d{12}$/.test(r)) {
+                logger.error(`${r} is not a valid account id`);
+                process.exit(1);
+            }
+        });
```

Comment:
If instead of using forEach you could use filter and error on any non empty array. Could report all invalid account id's instead of just one at a time.
Something like:
```suggestion
        const badAccounts = resources.filter(r => !/^\d{12}$/.test(r));
        if (badAccounts) {
            logger.error(`${badAccounts} are not a valid account ids`);
            process.exit(1);
        }
```

I tried to do `resources.filter(/d/.test)` directly but doesn't seem like that is allowed.

Reply:
I find that returning all possible concurrent errors better than one at a time when possible, but maybe personal preference.